src/Gui/Viewer/Viewer.cpp

```diff
@@ -229,6 +229,7 @@ void Viewer::startRendering( const Scalar dt ) {
 void Viewer::swapBuffers() {
     if ( isExposed() ) { m_context->swapBuffers( this ); }
     doneCurrent();
+    emit renderCompleted();
```

Comment:
From my previous experiences : Such a signal need to be decorrelated from Qt because it could be usefull to execute some core process (Engine or App).

- One example could be: free the memory that are living for only one frame.
I think there are some another examples... specially when building applications. 

So the observable pattern seems to be a good idea as it stands. Other events management could be interesting i don't know now.